### PR TITLE
FIX: TwinCAT plugin / blark and pytmc integration

### PR DESCRIPTION
**Pull request by klauer**
## Description
* Builds on #166 as blark is a dev dependency (or a "full" installation dependency)
* Fixes whatrecord TwinCAT integration, providing EPICS record name to PLC symbol name mappings and "go to source code"-style functionality

## Motivation and Context
* As noted in #166  / https://github.com/pcdshub/pcds-envs/issues/298 whatrecord's TwinCAT integration is currently non-functional
* One of the goals was to move away from the pytmc project parser for certain things. Going forward, we'll still need pytmc for the following:
   * NC settings
   * TMC parsing (and record generation)
* Aspects of the pytmc plugin from whatrecord were also moved into blark, as they didn't really fit here

## How Has This Been Tested?
* This adds a test suite for previously uncovered "extension" code

## Screenshots (if appropriate):
image

image

### CODE REVIEW

**Comment by klauer** [comment]
OK, so the frontend changes are going to have to wait for #162 
A bit of context:
* That PR is a massive frontend refactor I dropped on the ground a while back, so it doesn't make sense to fix up master
* Once this PR gets merged, I'll push fixes for the plugins to that PR and then we'll look at merging it (soonish?)

File: whatrecord/tests/blark_root
Comment (tangkong) [inline whatrecord/tests/blark_root:1]: If I understand this correctly, you're doing this primarily to grab projects from `blark` for the test suite?

**Review by tangkong — APPROVED** [review]
I'll green stamp this to get the ball rolling.  I gave it a quick once over and things looked reasonable.  Required tests pass, which is an important step forward

File: whatrecord/tests/blark_root
Comment (klauer) [reply whatrecord/tests/blark_root:1]: Right, without the project it's next to impossible to test out the plugin - leading to the issue Zach noted in #166 

Adding in the TwinCAT code lets us have an example PLC project and a PLC library it uses
whatrecord then can automatically follow the steps here, from a user-provided PV name:
1. User provides PV name
2. whatrecord maps PV name to PLC project
3. blark digs into the dependency chain of the PLC project and finds that the symbol comes from a library
4. whatrecord links: PV name -> startup script -> database file -> PLC project symbol name -> PLC project source code -> library source code

**Review by ZLLentz — APPROVED** [review]
I'm :+1: on these fixes, no red flags
Some day we need to dive into this code so I can understand the overall project structure better

**Pull request merged**